Bitfield with hook: error: ‘X’ is not a member of ‘__hlt::Test’

The following example from the docs fails to compile/run:

```
module Test;

import spicy;

type X = enum { A = 1, B = 2 };

public type Foo = unit {
    f: bitfield(8) {
        x1: 0..3 &convert=X($$);
        x2: 4..7 &convert=X($$);
    } { print self.f.x1, self.f.x2; }
};


```
(https://docs.zeek.org/projects/spicy/en/latest/programming/parsing.html#bitfield)

Removing the `{ print self.f.x1, self.f.x2; }` hook works around the issue.

Output is as follows:
```
$ printf "\x80" | spicy-driver bitfield-doc.spicy 
/tmp/user/1000/__linker___98ddb03444c98a9d-586cce8bc690138a.cc:15:67: error: ‘X’ is not a member of ‘__hlt::Test’
   15 |     extern void __hook_Foo___on_f(const hilti::rt::Bitfield<Test::X, Test::X, ::hilti::rt::integer::safe<uint8_t>>& __dd, ::hilti::rt::ValueReference<Foo>& __self);
      |                                                                   ^
/tmp/user/1000/__linker___98ddb03444c98a9d-586cce8bc690138a.cc:15:76: error: ‘X’ is not a member of ‘__hlt::Test’
   15 |     extern void __hook_Foo___on_f(const hilti::rt::Bitfield<Test::X, Test::X, ::hilti::rt::integer::safe<uint8_t>>& __dd, ::hilti::rt::ValueReference<Foo>& __self);
      |                                                                            ^
/tmp/user/1000/__linker___98ddb03444c98a9d-586cce8bc690138a.cc:15:113: error: template argument 1 is invalid
   15 |     extern void __hook_Foo___on_f(const hilti::rt::Bitfield<Test::X, Test::X, ::hilti::rt::integer::safe<uint8_t>>& __dd, ::hilti::rt::ValueReference<Foo>& __self);
      |                                                                                                                 ^~
/tmp/user/1000/__linker___98ddb03444c98a9d-586cce8bc690138a.cc:15:113: error: template argument 2 is invalid
/tmp/user/1000/__linker___98ddb03444c98a9d-586cce8bc690138a.cc:16:72: error: ‘X’ is not a member of ‘__hlt::Test’
   16 |     extern void __hook_Foo___on_f_fb8e(const hilti::rt::Bitfield<Test::X, Test::X, ::hilti::rt::integer::safe<uint8_t>>& __dd, ::hilti::rt::ValueReference<Foo>& __self);
      |                                                                        ^
/tmp/user/1000/__linker___98ddb03444c98a9d-586cce8bc690138a.cc:16:81: error: ‘X’ is not a member of ‘__hlt::Test’
   16 |     extern void __hook_Foo___on_f_fb8e(const hilti::rt::Bitfield<Test::X, Test::X, ::hilti::rt::integer::safe<uint8_t>>& __dd, ::hilti::rt::ValueReference<Foo>& __self);
      |                                                                                 ^
/tmp/user/1000/__linker___98ddb03444c98a9d-586cce8bc690138a.cc:16:118: error: template argument 1 is invalid
   16 |     extern void __hook_Foo___on_f_fb8e(const hilti::rt::Bitfield<Test::X, Test::X, ::hilti::rt::integer::safe<uint8_t>>& __dd, ::hilti::rt::ValueReference<Foo>& __self);
      |                                                                                                                      ^~
/tmp/user/1000/__linker___98ddb03444c98a9d-586cce8bc690138a.cc:16:118: error: template argument 2 is invalid
/tmp/user/1000/__linker___98ddb03444c98a9d-586cce8bc690138a.cc:23:76: error: ‘X’ is not a member of ‘__hlt::Test’
   23 | extern void __hlt::Test::__hook_Foo___on_f(const hilti::rt::Bitfield<Test::X, Test::X, ::hilti::rt::integer::safe<uint8_t>>& __dd, ::hilti::rt::ValueReference<Foo>& __self) { __hlt::Test::__hook_Foo___on_f_fb8e(__dd, __self); }
      |                                                                            ^
/tmp/user/1000/__linker___98ddb03444c98a9d-586cce8bc690138a.cc:23:85: error: ‘X’ is not a member of ‘__hlt::Test’
   23 | extern void __hlt::Test::__hook_Foo___on_f(const hilti::rt::Bitfield<Test::X, Test::X, ::hilti::rt::integer::safe<uint8_t>>& __dd, ::hilti::rt::ValueReference<Foo>& __self) { __hlt::Test::__hook_Foo___on_f_fb8e(__dd, __self); }
      |                                                                                     ^
/tmp/user/1000/__linker___98ddb03444c98a9d-586cce8bc690138a.cc:23:122: error: template argument 1 is invalid
   23 | extern void __hlt::Test::__hook_Foo___on_f(const hilti::rt::Bitfield<Test::X, Test::X, ::hilti::rt::integer::safe<uint8_t>>& __dd, ::hilti::rt::ValueReference<Foo>& __self) { __hlt::Test::__hook_Foo___on_f_fb8e(__dd, __self); }
      |                                                                                                                          ^~
/tmp/user/1000/__linker___98ddb03444c98a9d-586cce8bc690138a.cc:23:122: error: template argument 2 is invalid
[error] spicy-driver: JIT compilation failed
```

--- a/hilti/toolchain/src/compiler/codegen/types.cc
+++ b/hilti/toolchain/src/compiler/codegen/types.cc
@@ -32,6 +32,13 @@ struct VisitorDeclaration : hilti::visitor::PreOrder<cxx::declaration::Type, Vis
     auto typeID(const Node& n) { return n.as<Type>().typeID(); }
     auto cxxID(const Node& n) { return n.as<Type>().cxxID(); }
 
+    result_t operator()(const type::Bitfield& n) {
+        for ( const auto& b : n.bits(true) )
+            addDependency(b.itemType());
+
+        return {};
+    }
+
     result_t operator()(const type::Struct& n, const position_t p) {
         assert(typeID(p.node));
 

--- a/tests/spicy/types/bitfield/convert-with-enum.spicy
+++ b/tests/spicy/types/bitfield/convert-with-enum.spicy
@@ -1,0 +1,16 @@
+# @TEST-EXEC: spicyc -j %INPUT
+#
+# @TEST-DOC: Make sure this compiles. Regression test for #1568.
+
+module Test;
+
+import spicy;
+
+type X = enum { A = 1, B = 2 };
+
+public type Foo = unit {
+    f: bitfield(8) {
+        x1: 0..3 &convert=X($$);
+        x2: 4..7 &convert=X($$);
+    } { print self.f.x1, self.f.x2; }
+};

